lib/src/models.dart

```diff
@@ -1,7 +1,9 @@
 /// Defines a set of possible states which a [DownloadTask] can be in.
 @pragma('vm:entry-point')
+@Deprecated('Use the DownloadStatus class instead')
```

Comment:
I do appreciate the work you've put to make the breaking changes not-so-breaking, but if it slows you down, feel free to abandon the backwards compatibility and write some MIGRATION.md document.

Speaking more generally, I'd like to add docs for this package using [docs.page](https://docs.page/). Here's an example documentation created with docs.page: https://patrol.leancode.co/

Reply:
Is that some kind of hosted [docusaurus](https://docusaurus.io)?

When pausing and canceling is working on all platforms I can continue on this.

Reply:
It's [docs.page](https://docs.page)